src/vcpkg/binarycaching.cpp

```diff
@@ -914,6 +915,251 @@ namespace
         bool m_interactive;
         bool m_use_nuget_cache;
     };
+    struct GHABinaryProvider : IBinaryProvider
+    {
+        GHABinaryProvider(const VcpkgPaths& paths,
+                          std::vector<std::string>&& read_prefixes,
+                          std::vector<std::string>&& write_prefixes)
+            : paths(paths), m_read_prefixes(std::move(read_prefixes)), m_write_prefixes(std::move(write_prefixes))
+        {
+            m_read_cache_key = get_environment_variable("VCPKG_GHA_READ_KEY").value_or("vcpkg");
+            m_write_cache_key = get_environment_variable("VCPKG_GHA_WRITE_KEY").value_or("vcpkg");
+
+            auto token = get_environment_variable("VCPKG_ACTIONS_RUNTIME_TOKEN")
+                             .value_or(get_environment_variable("ACTIONS_RUNTIME_TOKEN").value_or(""));
```

Comment:
What is the purpose of the `VCPKG_ACTIONS_RUNTIME_TOKEN`? Why not just look for the `ACTIONS_RUNTIME_TOKEN`? Additionally, it looks like `ACTIONS_RUNTIME_TOKEN` is necessary for this to work. If it was not set, it should emit a message error notifying the user of the requirement along with a link to a docs page containing instructions for how to export the token and an example.